### PR TITLE
WP Block Styles: Only load in the editor if a theme opts in

### PR DESCRIPTION
**Pull request by scruffian**
## Description
The current behaviour is to always load opinionated block styles regardless of whether the theme uses them. This is a safe fall back for cases where a theme doesn't have any editor styles - so that the blocks in the editor don't look totally broken. However it has the drawback of overriding editor styles provided by the theme, meaning that theme editor styles end up becoming larger and more complex than necessary, to override these rules.

This PR changes that behaviour so that:
- For themes that opt into wp-block-styles, we always load the opinionated block styles (Independent Publisher 2)
- For themes that don't opt into wp-block-styles and don't provide editor styles, we always load the opinionated block styles (Kaura)
- For themes that don't opt into wp-block-styles and provide editor styles we don't load the opinionated block styles (Libre 2)

## How has this been tested?
With Independent Publisher 2, Karuna Libre 2

## Types of changes
Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [-] My code follows the accessibility standards. 
- [-] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [-] I've included developer documentation if appropriate. 
- [-] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by paaljoachim** [comment]
@aristath Ari I am thinking this is something you might want to look closer at.

File: lib/client-assets.php [hunk 1]
```diff
@@ -368,18 +368,24 @@ function gutenberg_register_packages_styles( $styles ) {
...
+	$wp_edit_blocks_dependencies = array(
+		'wp-components',
+		'wp-editor',
+		'wp-block-library',
+		'wp-reusable-blocks',
+	);
+
+	global $editor_styles;
+	if ( ! is_array( $editor_styles ) || count( $editor_styles ) === 0 ) {
+		// Always include visual styles is no $editor_styles are declared, so the editor never appears broken.
```
Comment (jasmussen) [inline lib/client-assets.php:380]: Nice.

**Comment by jasmussen** [comment]
> - For themes that opt into wp-block-styles, we always load the opinionated block styles (Independent Publisher 2)
> - For themes that don't opt into wp-block-styles and don't provide editor styles, we always load the opinionated block styles (Kaura)
> - For themes that don't opt into wp-block-styles and provide editor styles we don't load the opinionated block styles (Libre 2)

Honestly, this seems like a good heuristic to me, and provides a good balance of the editor experience sans editor styles, and theme developer expectations.

I think this could use a code review, but otherwise, 👍 👍 from me!

File: lib/client-assets.php (hunk 1 above)
Comment (jffng) [inline lib/client-assets.php:380]:
```suggestion
		// Always include visual styles if no $editor_styles are declared, so the editor never appears broken.
```

**Comment by carolinan** [comment]
I agree that this would be an improvement.

**Review by youknowriad — APPROVED** [review]

**Comment by youknowriad** [comment]
This makes sense to me. I thought this was how it worked already.

**Pull request merged**